test/Connext.t.sol

```diff
@@ -0,0 +1,168 @@
+  }
+
+  function mockTokenResponses(address asset) internal {
+    bytes memory selector = abi.encodeWithSelector(IERC20.approve.selector);
+    vm.mockCall(asset, selector, abi.encode(true));
+  }
+}
+
+contract ConnextFactoryConstructor is ConnextFactoryBaseTest {
+  function test_Constructor() public {
+    assertEq(address(factory.CONNEXT()), connext, "connext");
+  }
+}
+
+contract ConnextFactoryDeploy is ConnextFactoryBaseTest {
+  function test_DeployRouter(address asset) public {
+    vm.assume(asset != weth && asset != address(vm));
+    mockTokenResponses(asset);
+
+    address router1 = factory.computeAddress(asset);
+    address router2 = factory.deploy(asset);
+
+    assertEq(router1, router2, "address mismatch");
+    assertTrue(router1.code.length > 0, "no code");
+  }
+
+  function test_RevertsIfAssetAlreadyDeployed(address asset) public {
+    vm.assume(asset != address(vm));
+    mockTokenResponses(asset);
+
+    factory.deploy(asset);
+    vm.expectRevert(stdError.lowLevelError);
+    factory.deploy(asset);
+  }
+}
+
+contract ConnextFactoryGetRouter is ConnextFactoryBaseTest {
+  function test_GetRouter(address asset) public {
+    // WETH is excluded since it's already deployed.
+    vm.assume(asset != address(vm) && asset != weth);
+    mockTokenResponses(asset);
+
+    address router = factory.getRouter(asset);
+    assertEq(router, address(0), "router1");
+
+    factory.deploy(asset);
+    router = factory.getRouter(asset);
+    assertTrue(router > address(0), "router2");
+  }
+
+  function test_IsDeployed(address asset) public {
+    // WETH is excluded since it's already deployed.
+    vm.assume(asset != weth);
+    mockTokenResponses(asset);
+
+    assertFalse(factory.isDeployed(asset), "deployed1");
+    factory.deploy(asset);
+    assertTrue(factory.isDeployed(asset), "deployed2");
+  }
+}
+
+contract ConnextFactoryComputeAddress is ConnextFactoryBaseTest {
+  function test_ComputeAddressEth() public {
+    address router = factory.computeAddress(weth);
+    assertEq(router, factory.ETH_ROUTER(), "router");
+  }
+
+  function test_ComputeAddressToken(address asset) public {
+    vm.assume(asset != weth && asset != address(vm));
+    mockTokenResponses(asset);
+
+    address router1 = factory.computeAddress(asset);
+    address router2 = factory.deploy(asset);
+    assertEq(router1, router2, "router");
+  }
+
+  function test_ComputeAddressWeth() public {
+    address router = factory.computeAddress(weth);
+    assertEq(router, factory.ETH_ROUTER(), "router");
+  }
+}
+
+// =========================
+// ======== Routers ========
+// =========================
+
+contract RouterForkTestBase is Test {
+  // Optimism Goerli data, from https://docs.connext.network/resources/testnet.
+  IConnext connext = IConnext(0xA04f29c24CCf3AF30D4164F608A56Dc495B2c976);
+  address weth = 0x4E283927E35b7118eA546Ef58Ea60bfF59E857DB;
+  address tkn = 0x68Db1c8d85C09d546097C65ec7DCBFF4D6497CbF;
```

Comment:
What token is this? [TestToken](https://goerli-optimism.etherscan.io/token/0x68db1c8d85c09d546097c65ec7dcbff4d6497cbf#readContract)?

Reply:
Yea it's just a goerli testnet token that connext uses that lets  anyone mint an arbitrary amount to anyone, you can see it linked in their docs (that URL on L105) under "Production Testnet Contracts"